app/controllers/grade_entry_forms_controller.rb

```diff
@@ -253,27 +104,74 @@ def student_interface
   def get_mark_columns
     grade_entry_form = GradeEntryForm.find(params[:id])
     grade_entry_items_columns = grade_entry_form.grade_entry_items
-
     c = grade_entry_items_columns.map do |column|
       {
-        id: column.name,
+        id: column.id,
         content: column.name + ' (' + column.out_of.to_s + ')',
         sortable: true,
-        searcheable: true
+        compare: 'compare_gradebox'
       }
     end
     if grade_entry_form.show_total
       c <<
-      {
-        id: 'total_marks',
-        content: t('grade_entry_forms.grades.total') \
-                 + ' ' + grade_entry_form.out_of_total.to_s,
-      }
+        {
+          id: 'total_marks',
+          content: t('grade_entry_forms.grades.total') \
+                   + ' ' + grade_entry_form.out_of_total.to_s,
+          sortable: true,
+          compare: 'compare_gradebox'
+        }
+    end
+    if current_user.admin? || current_user.ta?
+      c <<
+        {
+          id: 'marking_state',
+          content: t('grade_entry_forms.grades.marking_state'),
+          sortable: true
+        }
     end
 
     render json: c
   end
 
+  def populate_grades_table
```

Comment:
Assignment Branch Condition size for populate_grades_table is too high. [38.13/15]